src/components/WorkoutSets/index.tsx

```diff
@@ -1,36 +1,210 @@
-import { CategoryColor } from '../../types/workout';
+import { CategoryColor, dayCombination, WorkoutItem } from '../../types/workout';
+
+import { Typography } from '@material-ui/core';
+import { DragDropContext, DropResult } from 'react-beautiful-dnd';
+import React from 'react';
+import { WorkoutSet, DayOfWeek } from '../../types/workout';
+import useWorkoutItems from '../WorkoutItems/hooks/useWorkoutItems';
+import axios from 'axios';
+import { WorkoutSetItemContext } from '../../store/WokroutSetItemCxt';
 
 type Props = {
   workoutsets: WorkoutSet[];
   workoutitems: WorkoutItem[];
   categorycolor: CategoryColor[];
 };
 
-const WorkoutSets: React.FC<Props> = ({
-  workoutsets,
-  workoutitems,
-  categorycolor,
-}: Props) => {
+type Data = {
+  workoutItemArray: WorkoutSet[][];
+  deleteIdList: number[];
+};
+
+const WorkoutSets: React.FC<Props> = ({ workoutsets, workoutitems, categorycolor }: Props) => {
+  let getDate = new Date();
+  let today = getDate.getMonth() + 1 + ' / ' + getDate.getDate() + ' / ' + getDate.getFullYear();
+  const [state, handlers] = useWorkoutItems(workoutitems);
+
+  let dayOfWeek = getDate.getDay();
+
+  const [dayOfToday, setDayOfToday] = React.useState(dayOfWeek);
+
+  let week: Array<WorkoutSet[]> = [[], [], [], [], [], [], []];
+
+  const [orderChangedWeek, setOrderChangedWeek] = React.useState(week);
+  const [deleteIdList, setDeleteIdList] = React.useState<number[]>([]);
+
+  console.log(deleteIdList);
+  React.useEffect(() => {
+    workoutsets.sort((a, b) => {
+      if (!a.set_order || !b.set_order) {
+        return -1;
+      }
+      return a?.set_order - b?.set_order;
+    });
+  }, [workoutsets]);
+  workoutsets.map((workoutset, i) => {
+    switch (workoutset.day_of_week) {
+      case DayOfWeek.Sun:
+        week[0].push(workoutset);
+        break;
+      case DayOfWeek.Mon:
+        week[1].push(workoutset);
+        break;
+      case DayOfWeek.Tue:
+        week[2].push(workoutset);
+        break;
+      case DayOfWeek.Wed:
+        week[3].push(workoutset);
+        break;
+      case DayOfWeek.Thu:
+        week[4].push(workoutset);
+        break;
+      case DayOfWeek.Fri:
+        week[5].push(workoutset);
+        break;
+      case DayOfWeek.Sat:
+        week[6].push(workoutset);
+        break;
+      default:
+        break;
+    }
+  });
+
+  const onPrevDayChangeHandler = () => {
+    setDayOfToday(dayOfToday === 0 ? 6 : dayOfToday - 1);
+  };
+
+  const onNextDayChangeHandler = () => {
+    setDayOfToday(dayOfToday === 6 ? 0 : dayOfToday + 1);
+  };
+
+  const onDeleteSetItem = React.useCallback((newWeekArr: WorkoutSet[][], deleteId: number | null) => {
+    if (deleteId) {
+      setDeleteIdList((prev) => [...prev, deleteId]);
+    }
+    setOrderChangedWeek([...newWeekArr]);
+  }, []);
+
+  const onDragEnd = (result: DropResult) => {
+    const { destination, source, draggableId } = result;
+
+    if (!destination) {
+      return;
+    }
```

Comment:
I think one liner is more readable
```suggestion
    if (!destination) return
```